src/Illuminate/Testing/TestResponse.php

```diff
@@ -1417,6 +1417,25 @@ public function assertSessionMissing($key)
         return $this;
     }
 
+    /**
+     * Assert that the Precognition request was successful
+     *
+     * @return $this
+     */
+    public function assertPrecognitionSuccess()
+    {
+        PHPUnit::assertTrue(
+            $this->headers->has('Precognition-Success'), 'Precognition-Success Header not present on response.'
+        );
+
+        PHPUnit::assertSame(
+            'True', $this->headers->get('Precognition-Success'),
+            'Precognition-Success Header was found, but the value is not `True`.'
+        );
```

Comment:
```suggestion
        $this->assertStatus(204);
        
        PHPUnit::assertTrue(
            $this->headers->has('Precognition-Success'), 'Precognition-Success Header not present on response.'
        );

        PHPUnit::assertSame(
            'true', $this->headers->get('Precognition-Success'),
            'Precognition-Success Header was found, but the value is not `true`.'
        );
```

Reply:
While yes, a Precognition response is, by default, a 204 status. This assertion should only look at the header. One of the reasons I was playing around with this was I wanted to see if I could customise the response to provide additional data in the event of a successful request. If someone wants to assert the status code, it should be done as part of the test, and keep this checking for the header.

Reply:
Hey Peter,

My concern with not asserting the status is that the precognition libraries check specifically for both a `204` status and `Precognition-Success` header - so a "Precognition Success" involves both of these in tandem.

https://github.com/laravel/precognition/blob/b4f47f7d933674771ef38d83fd2eca5c6f77620d/packages/core/src/client.ts#L18

I feel we need to replicate this check.

Regarding the functionality you are playing around with, you could return any response from a precognition request and respond to it in your application using the `onSuccess` configuration option. You don't need to include the `Precognition-Success` header.

```js
const form = useForm('post', '/users', {
    name: '',
    email: '',
}, {
    onSuccess(response) {
        // Will be invoked when the response is > 200 and < 300 
    },
});
```

Reply:
I meant making custom precognition Middleware/Controllers on the backend 😅 .

I've been working with customising the Middleware so specific routes can return small objects in response to a successful precognition response. At that point, I would change the response from a 204 to a 200.

Reply:
Maybe a good compromise would be to have an override boolean for the status? e.g.

```php
    public function assertPrecognitionSuccess(bool $assertStatus = true)
    {
         if ($assertStatus)
         {
             $this->assertNoContent();
         }
    }
```

Reply:
I totally see what you are trying to do but I would say that the the `204` and `Preocognition-Success` header is in place to indicate that the response got through all the user specific code  / middleware etc. and ended up back with the framework, i.e., no user returned response was intercepted.

What you are doing is totally valid and really cool to see people experiementing. I imagine it is something like...

```php
class MyMiddleware
{
    public function handle($request, $next)
    {
        if ($request->isPrecognitive()) {
            return response(['custom' => 'data']);
        }

        return $next($response);
    }
}
```

In this case I would argue that this response should not have a `"Precognition-Success"` header and is just a response returned for Precognitive requests.

I probably need to document the "Precognition Specification" to make this all clearer.

What you are doing is awesome, I just don't think you need to add the "Precognition-Success" header to the response.

I am totally keen for you to push back on all and any of this. I don't want to be shutting this down. Here to learn more if you feel there is something I am missing ❤️

Reply:
That all seems a fair enough explanation 🙂 I'll add the status assertion in.